src/api/wayfire/signal-definitions.hpp

```diff
@@ -114,9 +115,22 @@ struct input_event_signal : public wf::signal_data_t
      *
      * This is currently supported for only a subset of signals, namely:
      *
-     * pointer_button, keyboard_key, touch_down
+     * pointer_button, keyboard_key, touch_down,
+     * pointer_swipe_begin, pointer_swipe_update, pointer_swipe_end,
+     * pointer_pinch_begin, pointer_pinch_update, pointer_pinch_end,
+     * pointer_hold_begin, pointer_hold_end
      */
     input_event_processing_mode_t mode = input_event_processing_mode_t::FULL;
+
+    /**
+     * Whether the event has already been handled.
+     *
+     * This is useful for e.g. preventing conflicts between different plugins
+     * handling touchpad gestures.
+     *
+     * Setting this flag also implicitly sets the mode to NO_CLIENT.
+     */
+    bool carried_out = false;
```

Comment:
As we already have an input processing mode, I was wondering whether it won't be simpler to add an additional processing mode `IGNORE` ?

Reply:
maybe. But this is more consistent with other signals and possibly safer against unintentional modification in other places (since it's a new flag with only this meaning)?

Reply:
Looking at this again, I think you are right. Anyway, I think it is better to remove the `implicitly sets ...` bit, plugins should do that if they so desire, but there is no possibility for opt-out as it is currently.

Reply:
Looking at this a third time, I believe extending the input processing mode is better. I'll send a PR for that as I need it for something else (pointer buttons), and then if you still need this feature, feel free to send a new PR with the new gestures and handling for the gestures themselves. The relevant refactor is already over, so a potential PR with such changes could be merged now.

Reply:
FTR: #1804

Reply:
Hah, sorry, I haven't been following the project lately… sometime in the future I might come back…